pyproject.toml

```diff
@@ -1,52 +1,89 @@
-    "Harold Solbrig",
-    "Sujay Patil <spatil@lbl.gov>",
-    "Harshad Hegde",
-    "Mark Andrew Miller <MAM@lbl.gov>",
-    "Deepak Unni",
-    "Gaurav Vaidya <gaurav@renci.org>",
-    "Kevin Schaper <kevin@tislab.org>",
+  {name = "Chris Mungall", email = "cjmungall@lbl.gov"},
+  {name = "Sierra Moxon", email = "smoxon@lbl.gov"},
+  {name = "Harold Solbrig", email = "solbrig@jhu.edu"},
+  {name = "Sujay Patil", email = "spatil@lbl.gov"},
+  {name = "Harshad Hegde", email = "hhegde@lbl.gov"},
+  {name = "Mark Andrew Miller", email = "MAM@lbl.gov>"},
+  {name = "Deepak Unni"},
+  {name = "Gaurav Vaidya", email = "gaurav@renci.org>"},
+  {name = "Kevin Schaper", email = "kevin@tislab.org>"},
 ]
-
-readme = "README.md"
-
-homepage = "https://linkml.io/linkml/"
+license = "Apache-2.0"
+license-files = ["LICENSE"]
+homepage = "https://linkml.io/"
 repository = "https://github.com/linkml/linkml"
 documentation = "https://linkml.io/linkml/"
-
+readme = "README.md"
 keywords = ["schema", "linked data", "data modeling", "rdf", "owl", "biolink"]
-
 classifiers = [
     "Development Status :: 5 - Production/Stable",
     "Environment :: Console",
     "Intended Audience :: Developers",
     "Intended Audience :: Science/Research",
     "Intended Audience :: Healthcare Industry",
-    "License :: OSI Approved :: Apache Software License",
     "Topic :: Software Development :: Libraries :: Python Modules",
-    "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
 ]
-
-packages = [
-    { include = "linkml" }
+include = ["linkml"]
+requires-python = ">=3.9"
+dynamic = ["version"]
+dependencies = [  # Specifier syntax: https://peps.python.org/pep-0631/
+    "antlr4-python3-runtime >= 4.9.0, < 4.10",
+    "click >= 7.0",
+    "graphviz >= 0.10.1", 
+    "hbreader", 
+    "isodate >= 0.6.0",
+    "jinja2 >= 3.1.0",
+    "jsonasobj2 >= 1.0.3, < 2.0.0",
+    "jsonschema[format] >= 4.0.0",
+    "linkml-runtime >= 1.9.1, < 2.0.0",
+    "openpyxl",
+    "parse",
+    "prefixcommons >= 0.1.7",
+    "prefixmaps >= 0.2.2",
+    "pydantic >= 1.0.0, < 3.0.0",
+    "pyjsg >= 0.11.6",
+    "pyshex >= 0.7.20",
+    "pyshexc >= 0.8.3",
+    "python-dateutil",
+    "pyyaml",
+    "rdflib >=6.0.0",
+    "requests >= 2.22",
+    "sqlalchemy >= 1.4.31",
+    "watchdog >= 0.9.0",
+    "typing-extensions >= 4.6.0; python_version < '3.12'",
 ]
 
-[tool.poetry-dynamic-versioning]
-enable = true
-vcs = "git"
-style = "pep440"
+[project.optional-dependencies]
+black = [
+    "black >= 24.0.0",
+]
+numpydantic = [
+    "numpydantic >= 1.6.1",
+]
+shacl = [
+    "pyshacl >= 0.25.0",
+]
+tests = [
+    # "linkml[black]",  # recursive dependency not supported by poetry 
+    "black >= 24.0.0",
+    "numpydantic >= 1.6.1",
+    "pyshacl >= 0.25.0",
+]
```

Comment:
These changes do faithfully replicate what we previously had with `[tool.poetry.extras]`. But the `black` and `tests` groups don't really make sense to me. Those are development dependencies, and I'd expect to see them in `[tool.poetry.group.<group>.dependencies]` (for Poetry) or `[dependency-groups]` (for PEP 735-compatible tools). Anyway, no action needed now, but something to a closer look at as we continue to refine our project tooling.